components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeService.java

```diff
@@ -0,0 +1,373 @@
+     * @return Retrieved Scope
+     * @throws IdentityOAuth2ScopeException
+     */
+    public Scope getScopeByID(String scopeID) throws IdentityOAuth2ScopeException {
+        Scope scope;
+        Scope scopeFromCache = null;
+
+        if (scopeID == null) {
+            throw Oauth2ScopeUtils.handleClientException(Oauth2ScopeConstants.ErrorMessages.
+                    ERROR_CODE_BAD_REQUEST_SCOPE_ID, null);
+        }
+
+        if (OAuthServerConfiguration.getInstance().isCacheEnabled()) {
+            scopeFromCache = scopeCache.getValueFromCache(scopeID);
+        }
+        if (scopeFromCache == null) {
+            try {
+                scope = scopeMgtDAO.getScopeByID(scopeID, Oauth2ScopeUtils.getTenantID());
+                if (scope != null) {
+                    scopeCache.addToCache(scopeID, scope);
+                    if (log.isDebugEnabled()) {
+                        log.debug("Scope is added to the cache.");
+                    }
+                }
+
+            } catch (IdentityOAuth2Exception e) {
+                throw Oauth2ScopeUtils.handleServerException(Oauth2ScopeConstants.ErrorMessages.
+                        ERROR_CODE_FAILED_TO_GET_SCOPE_BY_KEY, null, e);
+            }
+        } else {
+            scope = scopeFromCache;
+            if (log.isDebugEnabled()) {
+                log.debug("Scope is getting from the cache.");
+            }
+        }
+
+        if (scope == null) {
+            throw Oauth2ScopeUtils.handleClientException(Oauth2ScopeConstants.ErrorMessages.
+                    ERROR_CODE_BAD_REQUEST_SCOPE_ID, null);
+        }
+
+        return scope;
+    }
+
+    /**
+     * Check the existence of a scope
+     *
+     * @param scopeName Name of the scope
+     * @return true if scope with the given scope name exists
+     * @throws IdentityOAuth2ScopeException
+     */
+    public boolean isScopeExists(String scopeName) throws IdentityOAuth2ScopeException {
+        String scopeID;
+        boolean isScopeExists = false;
+
+        if (scopeName == null) {
+            throw Oauth2ScopeUtils.handleClientException(Oauth2ScopeConstants.ErrorMessages.
+                    ERROR_CODE_BAD_REQUEST_SCOPE_NAME, null);
+        }
+        try {
+            scopeID = scopeMgtDAO.getScopeIDByName(scopeName, Oauth2ScopeUtils.getTenantID());
+        } catch (IdentityOAuth2Exception e) {
+            throw Oauth2ScopeUtils.handleServerException(Oauth2ScopeConstants.ErrorMessages.
+                    ERROR_CODE_FAILED_TO_GET_SCOPE_BY_KEY, null, e);
+        }
+
+        if (scopeID != null) {
+            isScopeExists = true;
+        }
+
+        return isScopeExists;
+    }
+
+    /**
+     * Delete the scope for the given scope ID
+     *
+     * @param scopeID Scope ID of the scope which need to get deleted
+     * @throws IdentityOAuth2ScopeException
+     */
+    public void deleteScopeByID(String scopeID) throws IdentityOAuth2ScopeException {
```

Comment:
add both IdentityOAuth2ScopeClinetException and IdentityOAuth2ScopeServerException